test/bufconn/bufconn_test.go

```diff
@@ -197,3 +197,103 @@ func TestCloseWhileAccepting(t *testing.T) {
 		t.Fatalf("c, err = %v, %v; want nil, %v", c, err, errClosed)
 	}
 }
+
+func TestDeadline(t *testing.T) {
+	sig := make(chan struct{})
+	var blockingWrite = func(conn net.Conn) {
+		conn.Write([]byte("0123456789"))
+		sig <- struct{}{}
+	}
+
+	var blockingRead = func(conn net.Conn) {
+		conn.Read(make([]byte, 10))
+		sig <- struct{}{}
+	}
+
+	p1, p2 := newPipe(5), newPipe(5)
+	c1, c2 := &conn{p1, p2}, &conn{p1, p2}
```

Comment:
Shouldn't `c2` be `&conn{p2, p1}`?

Reply:
Then a simultaneous `blockinRead(c1)` and `blockingWrite(c2)` (which I removed from the test) would not be blocking. Might be more clear to use `c1, c2 := &conn{p1, p1}, &conn{p2, p2}` though.